src/ng1/angular1.ts

```diff
@@ -80,21 +83,41 @@ function ng1UIRouter($locationProvider) {
     };
 
     bindFunctions(["replace", "url", "path", "search", "hash"], $location, services.location);
-    bindFunctions([ 'port', 'protocol', 'host'], $location, services.locationConfig);
+    bindFunctions(['port', 'protocol', 'host'], $location, services.locationConfig);
     bindFunctions(['baseHref'], $browser, services.locationConfig);
 
     return router;
   }
 }
 
+function resolveProvider() {
+  this.$get = $get;
+  function $get() {
+    return {
+      resolve: (invocables, locals, parent, self) => {
+        let context = new ResolveContext([]);
+        let resolvables: Resolvables = {};
+        forEach(invocables, (invocable, key) => {
+          resolvables[key] = new Resolvable(`${key}`, invocable);
+        });
+
+        context.addResolvables(resolvables, new State());
+
+        return context.resolvePath();
+      }
+    }
+  }
+}
+
 angular.module('ui.router.init', []).provider("ng1UIRouter", <any> ng1UIRouter);
 // Register as a provider so it's available to other providers
 angular.module('ui.router.util').provider('$urlMatcherFactory', ['ng1UIRouterProvider', () => router.urlMatcherFactory]);
 angular.module('ui.router.router').provider('$urlRouter', ['ng1UIRouterProvider', () => router.urlRouterProvider]);
 angular.module('ui.router.state').provider('$state', ['ng1UIRouterProvider', () => router.stateProvider]);
+angular.module('ui.router.resolve', []).provider('$resolve', <any> resolveProvider);
 
 /* This effectively calls $get() to init when we enter runtime */
 angular.module('ui.router.init').run(['ng1UIRouter', function(ng1UIRouter) { }]);
+angular.module('ui.router.resolve').run(['$resolve', function(resolve) { }]);
```

Comment:
this shouldn't be necessary since there is no initialization in the $resolve service.